### PR TITLE
Add examples for the glob crate

### PR DESCRIPTION
**Pull request by portal-chan**
Added both #218 and #219.

One thing I'm not too sure about is whether to use 

```rust
println!("{}", entry?.display());
```
or
```rust
if let Ok(path) = entry {
    println!("{}", path.display());
}
```

### CODE REVIEW

**Comment by budziq** [comment]
@portal-chan thanks! could you check why travis build is failing?

**Comment by expenses** [comment]
You might want to explain what the `error_chain!` macro does, as someone who hadn't used the `glob` crate yet (like me) might not be able to work outs its significance.

**Comment by budziq** [comment]
hi @expenses 
> You might want to explain what the error_chain! macro does, 

I would not refocus this example to error-chain crate. Especially as the `error_chain!`and rest of the error handling boiler plate will be hidden by default.

There explanations aimed for `error_chain` will be provided in their own separate examples
- https://github.com/brson/rust-cookbook/issues/214 - super easy
- https://github.com/brson/rust-cookbook/issues/215 - intermediate (reflecting the typical boilerplate for other examples)
Would you be willing to learn about `error_chain` crate while providing the example (with mentoring)

**Comment by portal-chan** [comment]
Other than me attempting to use git resulting in a disaster as usual the build is fine now. Sorry for all that.

**Comment by budziq** [comment]
> Also cargo test locally does not seem to work on windows due to it not being able to compile unix_socket.

Argh! This is due to `syslog` crate example which is only Linux compatible (which I've already fixed in rust-cookbook master https://github.com/brson/rust-cookbook/commit/83edf745826b80c7bc178e592ab767915fc25b8e). Unfortunately cursory test on my win box shows also a problem with our test runner skeptic on windows I've opened issue in their repo https://github.com/brson/rust-skeptic/issues/34 but it may take some time untill it's fixed and available on crates.io). 

Sorry about that! But hey we've got our first Windows contributor:tada:! 

In regard to 
> Other than me attempting to use git resulting in a disaster as usual the build is fine now. Sorry for all that.

Nothing to be sorry about! git has a learning curve and I know few people who work with it on daily basis that still make mistakes :wink:. The best way to get hang of it is to use the cli a few times (git guis are horrific abominations)

Please try to use `git rebase -i` to squash the 4 commits into one and push with `git push origin --force` to overwrite your PR. I'll review the PR in a moment.

File: src/basics.md [hunk 1]
```diff
@@ -412,6 +414,80 @@ fn run() -> Result<()> {
...
+# quick_main!(run);
+```
+
+[ex-glob-with]: #ex-glob-with
+<a name="ex-glob-with"></a>
+## Find all files with given pattern ignoring filename case.
+
+[![glob-badge]][glob] [![cat-filesystem-badge]][cat-filesystem]
+
+Find all image files in the `/media/` directory matching the `img_[0-9]*.png` pattern using the [glob] crate.
```
Comment (budziq) [inline src/basics.md]: I would suggest dropping the `using glob crate part`. Reader should be already informed about it thanks to badges/code snippet and other links in the description.

File: src/basics.md [hunk 2]
```diff
@@ -412,6 +414,80 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-glob-recursive]: #ex-glob-recursive
+<a name="ex-glob-recursive"></a>
+## Find all png files recursively
+
+[![glob-badge]][glob] [![cat-filesystem-badge]][cat-filesystem]
+
+Recursively find all png files using the [glob] crate.
```
Comment (budziq) [inline src/basics.md]: I would suggest dropping the `using glob crate part`. Reader should be already informed about it thanks to badges/code snippet and other links in the description. Honestly the surrounding examples should be also corrected to follow this rule.

**Comment by portal-chan** [comment]
@budziq squashed the commits thanks to your help.

Now I moved the examples to `app.md`, made them `no_run`, clarified the `**` pattern thing and made them point to the latest docs. Think that's about it.

**Comment by budziq** [comment]
Well done @portal-chan :+1: !

I just squashed the commits into single one prior to merging. 
You can see your examples deployed https://brson.github.io/rust-cookbook/app.html#ex-glob-recursive

**Pull request closed without merging**